### PR TITLE
[DX-369]use show parameter hide red lines and add config param to hide them

### PR DESCRIPTION
**Pull request by yurisasuke**
## Description
We should be able to hide and show the red lines in menu items that do not have a path. I have added a value in showEmptyPages that can be used to show or hide the red lines. I also use the .show value in the directory and the .show value in the pages menu to hide and show these red lines.
https://deploy-preview-2691--tyk-docs.netlify.app/docs/nightly/tyk-stack/

## Preview Link
https://deploy-preview-2691--tyk-docs.netlify.app/docs/nightly/tyk-stack/

## Types of changes

- [ ] Fixing typo (please merge to production)
- [ ] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [ ] Something else (please add if needs merging to production or not)

## Checklist

- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I have added a preview link.

### CODE REVIEW

File: scripts/menu_generator.py [hunk 1]
```diff
@@ -331,11 +331,52 @@ def print_tree_as_yaml(tree, level=1):
...
+    """
+    found_path = False
+
+    for menu_node in nodeList:
+        menu_category = menu_node.get("category")
+        children = menu_node.get("children", [])
+
+        if menu_category == "Page":
+            path = menu_node.get("url")
+            if path is not None:
```
Comment (letzya) [inline scripts/menu_generator.py]: test for length

File: scripts/menu_generator.py [hunk 2]
```diff
@@ -331,11 +331,52 @@ def print_tree_as_yaml(tree, level=1):
...
+                menu_node["show"] = True
+                found_path = True
+            else:
+                menu_node["show"] = False
+        elif menu_category == "Directory":
+            menu_node["show"] = process_show_status(children)
+            if menu_node.get("show"):
+                found_path = True
+        elif menu_category == "Tab":
```
Comment (letzya) [inline scripts/menu_generator.py:365]: also tabs can be hidden

**Review by letzya — CHANGES_REQUESTED** [review]
improve per comments

File: scripts/menu_generator.py (hunk 1 above)
Comment (yurisasuke) [reply scripts/menu_generator.py]: Done

File: scripts/menu_generator.py (hunk 2 above)
Comment (yurisasuke) [reply scripts/menu_generator.py:365]: Done

**Review by dcs3spp — APPROVED** [review]
PR looks good to me Keith

**Pull request merged**